### ISSUE TITLE
Constructor override is out-of-sync with Ruby 3.3.0's Logger class

### ISSUE THREAD

**Issue by st0012** [steveklabnik/mono_logger#13]
After https://github.com/ruby/logger/pull/85, `Logger#initialize` creates a `@level_override = {}` and access it [every time `level` is called ](https://github.com/ruby/logger/blob/master/lib/logger.rb#L588). 

Since `MonoLogger` overrides the constructor but doesn't initialises the same `@level_override` variable, it now raises `NoMethodError` whenever `level` is called.

```
irb(main):001:0> require "mono_logger"
=> true
irb(main):002:0> MonoLogger.new(nil).level
/opt/rubies/3.3.0-dev/lib/ruby/3.3.0+0/logger.rb:385:in `level': undefined method `[]' for nil (NoMethodError)
                                                                                          
    @level_override[Fiber.current] || @level                                              
                   ^^^^^^^^^^^^^^^                                                        
        from (irb):2:in `<main>'                                                          
        from <internal:kernel>:187:in `loop'                                              
```

**Comment by st0012** [comment]
Should be fixed by #12

**Comment by st0012** [comment]
Closing as #12 has been merged
